src/main/java/com/avioconsulting/mule/vault/provider/internal/connection/provider/AppRoleConnectionProvider.java

```diff
@@ -0,0 +1,67 @@
+package com.avioconsulting.mule.vault.provider.internal.connection.provider;
+
+import com.avioconsulting.mule.vault.provider.api.connection.parameters.TlsContext;
+import com.avioconsulting.mule.vault.provider.internal.connection.VaultConnection;
+import com.avioconsulting.mule.vault.provider.internal.connection.impl.AppRoleConnection;
+import com.avioconsulting.mule.vault.provider.internal.connection.impl.TokenConnection;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.meta.ExpressionSupport;
+import org.mule.runtime.config.api.dsl.model.ConfigurationParameters;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.Expression;
+import org.mule.runtime.extension.api.annotation.dsl.xml.ParameterDsl;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.display.DisplayName;
+import org.mule.runtime.extension.api.annotation.param.display.Placement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@DisplayName("AppRole Connection")
+@Alias("approle-connection")
+public class AppRoleConnectionProvider extends AbstractConnectionProvider {
+    private static final Logger logger = LoggerFactory.getLogger(AppRoleConnectionProvider.class);
+
+    @DisplayName("AppRole Path")
```

Comment:
Change name to provide more clarity that this is the path to the authentication provider

Reply:
Changed to AppRole Mount and added a summary to clarify